pkg/apis/internal.admin.acorn.io/v1/default.go

```diff
@@ -44,35 +56,70 @@ func getCurrentProjectComputeClassDefault(ctx context.Context, c client.Client,
 		return projectComputeClasses.Items[i].Name < projectComputeClasses.Items[j].Name
 	})
 
-	var defaultPCC *ProjectComputeClassInstance
+	var defaultPCC, projectSpecifiedPCC *ProjectComputeClassInstance
 	for _, projectComputeClass := range projectComputeClasses.Items {
 		if projectComputeClass.Default {
 			if defaultPCC != nil {
 				return nil, fmt.Errorf(
 					"cannot establish defaults because two default computeclasses exist: %v and %v",
 					defaultPCC.Name, projectComputeClass.Name)
 			}
-			t := projectComputeClass // Create a new variable that isn't being iterated on to get a pointer
-			defaultPCC = &t
+
+			// Create a new variable that isn't being iterated on to get a pointer
+			defaultPCC = z.Pointer(projectComputeClass)
+		}
+
+		if projectSpecified != "" && projectComputeClass.Name == projectSpecified {
+			projectSpecifiedPCC = z.Pointer(projectComputeClass)
 		}
 	}
 
+	if projectSpecifiedPCC != nil {
+		return projectSpecifiedPCC, nil
+	}
+
 	return defaultPCC, nil
 }
 
-func GetDefaultComputeClass(ctx context.Context, c client.Client, namespace string) (string, error) {
-	pcc, err := getCurrentProjectComputeClassDefault(ctx, c, namespace)
+// GetDefaultComputeClassName gets the name of the effective default ComputeClass for a given project namespace.
+// The precedence for picking the default ComputeClass is as follows:
+//  1. Any ProjectComputeClassInstance (in the project namespace) or ClusterComputeClassInstance that is specified by the
+//     ProjectInstance's DefaultComputeClass field
+//  2. The ProjectComputeClassInstance (in the project namespace) with a Default field set to true
+//  3. The ClusterComputeClassInstance with a Default field set to true
+//
+// If no default ComputeClass is found, an empty string is returned.
+func GetDefaultComputeClassName(ctx context.Context, c client.Client, namespace string) (string, error) {
+	var project internalv1.ProjectInstance
+	if err := c.Get(ctx, client.ObjectKey{Name: namespace}, &project); err != nil {
+		return "", fmt.Errorf("failed to get projectinstance to determine default compute class: %w", err)
+	}
+	projectSpecified := project.Status.DefaultComputeClass
+
+	var defaultComputeClasses []string
+	pcc, err := getCurrentProjectComputeClassDefault(ctx, c, projectSpecified, namespace)
 	if err != nil {
 		return "", err
-	} else if pcc != nil {
-		return pcc.Name, nil
+	}
+	if pcc != nil {
+		defaultComputeClasses = append(defaultComputeClasses, pcc.Name)
 	}
 
-	ccc, err := getCurrentClusterComputeClassDefault(ctx, c)
+	ccc, err := getCurrentClusterComputeClassDefault(ctx, c, projectSpecified)
 	if err != nil {
 		return "", err
-	} else if ccc != nil {
-		return ccc.Name, nil
 	}
+	if ccc != nil {
+		defaultComputeClasses = append(defaultComputeClasses, ccc.Name)
+	}
+
+	if sets.New(defaultComputeClasses...).Has(projectSpecified) {
```

Comment:
nit: slightly more performant.
```suggestion
	if slices.Contains(defaultComputeClasses, projectSpecified) {
```